Tensile/KernelWriterAssembly.py

```diff
@@ -10347,13 +10396,15 @@ def globalWriteElements(self, kernel, vectorWidths, elements,
       #jgolds look at moving these converted values back to scalar regs and free up the VGPRs
       # ethan if using (h,h,h,h,s,s) + HPA then the host should pass in an F32 alpha, we don't have to do the cvt
       if beta:
-        self.betaVgpr = self.vgprPool.checkOut(1, "beta")
-        kStr += inst("v_mov_b32", vgpr(self.betaVgpr), sgpr("Beta"), "sgpr -> vgpr b/c op_sel")
-        kStr += inst("v_cvt_f32_f16", vgpr(self.betaVgpr), vgpr(self.betaVgpr), "convert beta to fp32")
-        if self.betaInSgpr:
-          kStr += inst("v_readfirstlane_b32", sgpr("Beta"), vgpr(self.betaVgpr), "restore beta sgpr")
-          self.vgprPool.checkIn(self.betaVgpr)
-          self.betaVgpr = None
+        if kernel["ProblemType"]["HighPrecisionAccumulate"] and kernel["ProblemType"]["DestDataType"].isHalf():
+          self.betaVgpr = self.vgprPool.checkOut(1, "beta")
+          kStr += inst("v_mov_b32", vgpr(self.betaVgpr), sgpr("Beta"), "sgpr -> vgpr b/c op_sel")
+          kStr += inst("v_cvt_f32_f16", vgpr(self.betaVgpr), vgpr(self.betaVgpr), "convert beta to fp32")
+          if self.betaInSgpr:
+            kStr += inst("v_readfirstlane_b32", sgpr("Beta"), vgpr(self.betaVgpr), "restore beta sgpr")
+            self.vgprPool.checkIn(self.betaVgpr)
+            self.betaVgpr = None
```

Comment:
The outer "if" already checked if the kernel is <H/H/H>+HPA and <H/S/H>+HPA, so
1. No need to check kernel["ProblemType"]["HighPrecisionAccumulate"] again here
2. Why does the beta conversion requires kernel["ProblemType"]["DestDataType"].isHalf() ?  I think we still need to do this conversion once ComputeDataType is Half + HPA. So maybe we also don't need this DestDataType.isHalf test as well ?

Reply:
Hm I think this is a merge artifact from #1169 and what I needed before.

- Fixed.